### DISCUSSION TITLE
Generate roadmaps with AI

### DISCUSSION THREAD

**Discussion by BenutzerEinsZweiDrei** [learn-awesome/learndb#79] [Ideas]
I just created https://benutzereinszweidrei.github.io/ai-paths.html

Took me like a few hours

And then i tried to scan your database but it said you would need an api

Made me think that you guys could create roadmaps yourself with scanning your own database with AI

Could you do that so there is self learning study paths or roadmaps for every university degree?

You could still put a signal if a roadmap is generated or maintained by a professional to show its quality

**Comment by BenutzerEinsZweiDrei** [comment]
This should be easily possible with https://github.com/pashpashpash/vault-ai

**Comment by nileshtrivedi** [comment]
Hi @BenutzerEinsZweiDrei The page at https://benutzereinszweidrei.github.io/ai-paths.html does not seem to work.

**Comment by BenutzerEinsZweiDrei** [comment]
Hey yes. Im just coding stuff for myself privately as a hobby. Im not always considering someone is seeing it lol. So i changed the repository name and so the github pages address changed. You can still find it on my profile as "open learning paths".

I now found a website where you can train a chatbot on data and uploaded your database. You can ask him questions about it now. (Sorry lol you cant. Apparently i exceeded the limit on the free tier alrdy) https://bit.ly/4aSYKYg  (maybe its easier to upload it as csv to https://www.chatcsv.co/ to talk with it)

But it doesnt work good at all. Apparently there is a difference if you train him on data or if you give him the data as context. 

Vault AI seems to create a vector database and feed each prompt with the database as context. But since it is using OpenAI and the prompts with context consume insane amounts of tokens: this means using Vault AI with rly big databases costs a lot of money.

So maybe it only makes sense to just generate some roadmaps once. And in that usecase the quality would be better if you do it yourself by hand and not trust on the AI results.

**Comment by BenutzerEinsZweiDrei** [comment]
One possible solution would be

- fork vault ai
- make it use mistral or llama instead of OpenAI
- make it use chroma or weaviate instead of Pinecone
- host all free of cost privately
- feed it with your database
- make api calls to vault ai

Sadly i dont code good enough for this.

**Comment by nileshtrivedi** [comment]
Using AI to generate self-learning study paths seems vulnerable to influencer marketing which has polluted the web for their mediocre courses and books. Getting recommendations from LLMs is basically a popularity contest. I'd recommend relying on human-curated paths for now.
